includes/internal/emails/class-email-user-settings.php

```diff
@@ -0,0 +1,181 @@
+/**
+ * File containing the Email_User_Settings class.
+ *
+ * @package sensei
+ */
+
+namespace Sensei\Internal\Emails;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Email_User_Settings
+ *
+ * @internal
+ *
+ * @since $$next-version$$
+ */
+class Email_User_Settings {
+
+	/**
+	 * The Email_Repository instance.
+	 *
+	 * @var Email_Repository
+	 */
+	private $repository;
+
+	/**
+	 * The constructor.
+	 *
+	 * @internal
+	 *
+	 * @param Email_Repository $repository The Email_Repository instance.
+	 */
+	public function __construct( Email_Repository $repository ) {
+		$this->repository = $repository;
+	}
+
+	/**
+	 * Initialize the class and add hooks.
+	 *
+	 * @internal
+	 */
+	public function init() {
+		// Show the opt-in/out email settings in the user profile page.
+		add_action( 'show_user_profile', [ $this, 'maybe_add_email_settings' ] );
+		add_action( 'edit_user_profile', [ $this, 'maybe_add_email_settings' ] );
+
+		// Save the user email opt-in/out settings.
+		add_action( 'personal_options_update', [ $this, 'save_user_email_opt_in_out_settings' ] );
+		add_action( 'edit_user_profile_update', [ $this, 'save_user_email_opt_in_out_settings' ] );
+
+		// Filter to determine if an email should be sent to a user.
+		add_filter( 'sensei_send_emails', [ $this, 'should_send_email_to_user' ], 10, 5 );
+	}
+
+	/**
+	 * Save user email opt-in/out settings.
+	 *
+	 * @internal
+	 *
+	 * @param int $user_id The user ID.
+	 */
+	public function save_user_email_opt_in_out_settings( $user_id ) {
+		if ( ! current_user_can( 'edit_user', $user_id ) ) {
+			return false;
+		}
+
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.NonceVerification.Missing -- Nonce verified before hook is called, Input is sanitized in the next lines.
+		$opt_in_out_emails = $_POST['sensei-email-subscriptions'] ?? [];
+
+		if ( ! is_array( $opt_in_out_emails ) ) {
+			return;
+		}
+
+		$opt_in_out_emails = array_map( 'wp_unslash', $opt_in_out_emails );
+		$opt_in_out_emails = array_map( 'sanitize_text_field', $opt_in_out_emails );
+
+		$user_emails = $this->get_emails_for_user( $user_id );
+
+		foreach ( $user_emails as $identifier => $email ) {
+			$should_subscribe = in_array( $identifier, $opt_in_out_emails, true );
+
+			if ( $should_subscribe ) {
+				delete_user_meta( $user_id, 'sensei_email_unsubscribed_' . $identifier );
+			} else {
+				update_user_meta( $user_id, 'sensei_email_unsubscribed_' . $identifier, 'yes' );
+			}
+		}
+	}
+
+	/**
+	 * Add opt-in/out setting fields in user profile page.
+	 *
+	 * @param \WP_User $profile_user The user object.
+	 *
+	 * @internal
+	 */
+	public function maybe_add_email_settings( $profile_user ) {
+		$user_emails = $this->get_emails_for_user( $profile_user->ID );
+
+		if ( empty( $user_emails ) ) {
+			return;
+		}
+
+		?>
+			<h3><?php esc_html_e( 'Sensei Email Subscriptions', 'sensei-lms' ); ?></h3>
```

Comment:
I think we can simplify to "Sensei Emails" so we also pick up existing translations.